### ISSUE TITLE
Bigquery dialect: Problem parsing ALTER TABLE ... ALTER COLUMN syntax

### ISSUE THREAD

**Issue by michalklempa** [sqlfluff/sqlfluff#4482]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Dropping a `NOT NULL` [constraint from a table column syntax](https://cloud.google.com/bigquery/docs/managing-table-schemas#make_a_column_nullable_in_an_existing_table) is not parsed correctly, ends up with error.

### Expected Behaviour

Linting:

```sql
ALTER TABLE students ALTER COLUMN student_name DROP NOT NULL
```

should not end up with parsing error.

### Observed Behaviour

```
== [a.sql] FAIL
L:   1 | P:  28 | L014 | Unquoted identifiers must be consistently lower case.
L:   1 | P:  35 |  PRS | Line 1, Position 35: Found unparsable section:
                       | 'student_name DROP NOT NULL'
```

I think the word `COLUMN` is matched as identifier currently, therefore also rule L014 fires.

### How to reproduce

Take

```sql
ALTER TABLE students ALTER COLUMN student_name DROP NOT NULL
```

and run 
```
sqlfluff lint --dialect bigquery <file>
```
on it.

### Dialect

bigquery

### Version

```
> sqlfluff --version
sqlfluff, version 1.4.5
```

```
> python --version
Python 3.8.6
```

### Configuration

default

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by yoichi** [comment]
I think this is already fixed by #4316

**Comment by greg-finley** [comment]
Yep, try the 2.x alpha releases, or the 2.0.0 release coming ~today.
